### ISSUE TITLE
problems updating connection

### ISSUE THREAD

**Issue by jordiruiz** [Microsoft/PowerBI-Cli#32]
Hello,

We have some problems when trying to update the connection on a direct query report. 
We have a job that checks for reports in a folder and then automatically upload to azure powerbi workspace using the nodejs powerbi client. 
The process check if exist the report in the workspace and if it exist delete it and then upload again. Sometimes we receive an error and if we try it again the command returns that the connection was updated succesfully
We send the command with a parametrized bat file that contains the following (removed private data):

`@echo off
setlocal

powerbi update-connection -w workspace -d %1 -u user -p password -s "data source=ourserver.database.windows.net;persist security info=True;Connection Timeout=120;encrypt=True;trustservercertificate=True;initial catalog=%2"`

In the following screenshot you can see the same command executed twice with diferent results:

![image](https://cloud.githubusercontent.com/assets/2950890/19573800/e5b22ffc-9707-11e6-8e33-6737dad60547.png)

When the command tells that the connection is changed, when we load the report we receive an error if we send the command again the report loads correctly.

The error when load the report: 

![image](https://cloud.githubusercontent.com/assets/2950890/19574116/272ab8f4-9709-11e6-803a-8474b265bb70.png)

**Comment by jordiruiz** [comment]
I've saw that when I change the connection successfully at the first time but receives the error DMTS_DatasourceHasNoCredentialError when trying to load the report. If I run the update command again the command result has a different datasource id and then the report is loaded correctly. Seems like at the first time is getting a wrong datasoure.

first time result --> DMTS_DatasourceHasNoCredentialError:

***
**[ powerbi ] Found 1 Datasources for Dataset 51XXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXc8
[ powerbi ] --------------------------------------------------------------------
[ powerbi ] Datesource ID:  39XXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXX52** 
***

second time result --> OK:

***
**[ powerbi ] Found 1 Datasources for Dataset 51XXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXc8
[ powerbi ] --------------------------------------------------------------------
[ powerbi ] Datesource ID:  57XXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXf2**
***

**Comment by ali-hamud** [comment]
Is this still the case?

**Comment by FutureGUIs** [comment]
Yep, I'm getting the same problem, the first time I run the update-connection command it fails with 
`{"error":{"code":"DMTS_MonikerNotFoundError","pbi.error":{"code":"DMTS_MonikerNotFoundError","parameters":{},"details":[]}}}`

But the second time around it works
